plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/review-rating/index.tsx

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { registerBlockType } from '@wordpress/blocks';
+import { isExperimentalBlocksEnabled } from '@woocommerce/block-settings';
+import { Icon, starHalf } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import edit from './edit';
+
+if ( isExperimentalBlocksEnabled() ) {
+	// @ts-expect-error metadata is not typed.
+	registerBlockType( metadata, {
+		edit,
+		icon: {
+			src: (
+				<Icon
+					icon={ starHalf }
+					className="wc-block-editor-components-block-icon"
+				/>
+			),
+		},
```

Comment:
FWIW, out of curiosity, I checked whether directly passing the icon works, and it seems that it does. I was just curious and looked at the [code in Gutenberg](https://github.com/WordPress/gutenberg/blob/1c698f8ccdcc8a86b91f9e7efef608ad66f3a46b/packages/blocks/src/api/utils.js#L94-L130).

```suggestion
		icon: (
			<Icon
				icon={ starHalf }
				className="wc-block-editor-components-block-icon"
			/>
		),
```

Reply:
Yeah even the use of `Icon` is not necessary, and we can just use `icon: starHalf`.
I don't quite remember doing this explicitly myself, so I may of copied this from the average rating block: https://github.com/woocommerce/woocommerce/blob/99f04562c084808e02515a526664fdc0c77fa688/plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/average-rating/index.tsx#L16

Reply:
Just passed icon directly: https://github.com/woocommerce/woocommerce/pull/56332/commits/1b60ec1bb62284ce683c744014552a7006ef5a77

Reply:
Nice 🚀